### PR TITLE
Feature/add prison filtering to content

### PR DESCRIPTION
**Pull request by psoleckimoj**

### CODE REVIEW

File: modules/custom/moj_resources/src/Utilities.php [hunk 1]
```diff
@@ -34,4 +38,88 @@ public static function filterByPrisonCategories($prisonId, $prisonCategories, $q
...
+  }
+
+  /**
+   * Get Prison Categories for a Drupal node object
+   *
+   * @param EntityInterface $term
+   * @return int[]
+  */
+  public static function getPrisonCategoriesFor($node) {
+    $prisonCategories = array();
```
Comment (psoleckimoj) [inline modules/custom/moj_resources/src/Utilities.php:109]: Amusingly, `$a = [];` is not the same as `$a = array();`. The former causes `count()` to return `1` due to 

> When the parameter is neither an array nor an object with implemented Countable interface, 1 will be returned

File: modules/custom/moj_resources/src/ContentApiClass.php [hunk 2]
```diff
@@ -24,79 +26,112 @@ class ContentApiClass
...
+   * Validate the content is OK for prison/prison categories
+   *
+   * @return NodeInterface
+   */
+  private function getMatchingContent() {
+    $prison = Utilities::getTermFor($this->prisonId, $this->termStorage);
+    $content = Utilities::getNodeFor($this->contentId, $this->nodeStorage);
+    $contentPrisons = Utilities::getPrisonsFor($content);
+
+    if (count($contentPrisons) === 0) {
```
Comment (LucasCairns) [inline modules/custom/moj_resources/src/ContentApiClass.php:84]: Could we use `empty($contentPrisons)`

File: modules/custom/moj_resources/src/Utilities.php (hunk 1 above)
Comment (LucasCairns) [reply modules/custom/moj_resources/src/Utilities.php:109]: Interesting, I'm assuming this is similar to in JS with `[1, 2, 3]` and `new Array(1, 2, 3)`?

File: modules/custom/moj_resources/src/ContentApiClass.php (hunk 2 above)
Comment (psoleckimoj) [reply modules/custom/moj_resources/src/ContentApiClass.php:84]: Yep, it seems so.

File: modules/custom/moj_resources/src/Utilities.php (hunk 1 above)
Comment (psoleckimoj) [reply modules/custom/moj_resources/src/Utilities.php:109]: It caused me a lot of swearing when I could see the array was empty when printed out but still didn't pass the count.

File: modules/custom/moj_resources/src/ContentApiClass.php (hunk 2 above)
Comment (psoleckimoj) [reply modules/custom/moj_resources/src/ContentApiClass.php]: @LucasCairns changed to empties...

File: modules/custom/moj_resources/src/Utilities.php (hunk 1 above)
Comment (LucasCairns) [reply modules/custom/moj_resources/src/Utilities.php:109]: I've also noticed similar issues with `Entity` objects returned by Drupal, they're iterable, but don't work with all Array functions

**Review by LucasCairns — APPROVED** [review]
👍

**Pull request merged**